rust/rbac-registration/src/cardano/cip509/mod.rs

```diff
@@ -92,7 +107,13 @@ pub(crate) enum Cip509IntIdentifier {
 impl Decode<'_, ()> for Cip509 {
     fn decode(d: &mut Decoder, ctx: &mut ()) -> Result<Self, decode::Error> {
         let map_len = decode_map_len(d, "CIP509")?;
-        let mut cip509_metadatum = Cip509::default();
+
+        let mut purpose = Uuid::default();
+        let mut txn_inputs_hash = TxInputHash::default();
```

Comment:
Maybe better to set "not set" value to `None`.
Or check whether these required fields are set or not, if no return Error.

Can think about it in other PR since it is not related to this PR .

Reply:
Agree, don't assume that the fields are some default if they are not decoded, they should be set to a "not set" state, typically by using `Option<>` and defaulting them to None.

Reply:
I totally agree, but previously the whole `Cip509` structure was constructed in the default state, so I wasn't sure if it is ok.